internal/k8s/reconciler/validator/gateway_test.go

```diff
@@ -0,0 +1,713 @@
+package validator
```

Comment:
These tests were moved over from `reconciler/gateway_test.go` and `reconciler/listener_test.go`